mpc/share-conversion-core/src/lib.rs

```diff
@@ -1,23 +1,200 @@
 //! This subcrate implements secure two-party (2PC) multiplication-to-addition (M2A) and
 //! addition-to-multiplication (A2M) algorithms, both with semi-honest security.
 //!
-//! ### M2A algorithm
+//! ### M2A algorithm (implementation of chapter 4.1 in <https://link.springer.com/content/pdf/10.1007/3-540-48405-1_8.pdf>)
 //! Let `A` be an element of some finite field with `A = a * b`, where `a` is only known to Alice
 //! and `b` is only known to Bob. A is unknown to both parties and it is their goal that each of
 //! them ends up with an additive share of A. So both parties start with `a` and `b` and want to
 //! end up with `x` and `y`, where `A = a * b = x + y`.
 //!
-//! ### A2M algorithm
+//! ### A2M algorithm (adaptation of chapter 4 in <https://www.cs.umd.edu/~fenghao/paper/modexp.pdf>)
 //! This is the other way round.
 //! Let `A` be an element of some finite field with `A = x + y`, where `x` is only known to Alice
 //! and `y` is only known to Bob. A is unknown to both parties and it is their goal that each of
 //! them ends up with a multiplicative share of A. So both parties start with `x` and `y` and want to
 //! end up with `a` and `b`, where `A = x + y = a * b`.
 
-pub mod gf2_128;
+mod a2m;
+pub mod fields;
+mod m2a;
+
+pub use a2m::AddShare;
+use fields::Field;
+pub use m2a::MulShare;
+use rand::{CryptoRng, Rng};
+
+/// A trait for converting field elements
+///
+/// Allows two parties to switch between additively and multiplicatively
+/// shared representations of a field element.
+pub trait ShareConvert: Copy
+where
+    Self: Sized,
+{
+    type Inner: Field;
+    type Output: ShareConvert<Inner = Self::Inner, Output = Self>;
+
+    /// Create a new instance
+    fn new(share: Self::Inner) -> Self;
+
+    /// Converts '&self' into choices needed for the receiver input to an oblivious transfer.
+    /// The choices are in the "least-bit-first" order.
```

Comment:
Either the comment is wrong or the code is wrong. The way I read the code is:
- take the MSB bit and push it to `out`
- repeat for the rest of the bits until LSB bit is pushed
the result is that the first bit of `out` is MSB. This is not "least-bit-first" as the comment says.